controller_interface/include/semantic_components/gps_sensor.hpp

```diff
@@ -0,0 +1,118 @@
+// Copyright 2025 ros2_control development team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef SEMANTIC_COMPONENTS__GPS_SENSOR_HPP_
+#define SEMANTIC_COMPONENTS__GPS_SENSOR_HPP_
+
+#include <string>
+#include <vector>
+
+#include "semantic_components/semantic_component_interface.hpp"
+#include "sensor_msgs/msg/nav_sat_fix.hpp"
+
+namespace semantic_components
+{
+class GPSSensor : public SemanticComponentInterface<sensor_msgs::msg::NavSatFix>
```

Comment:
```suggestion
template <bool covariance>
class GPSSensor : public SemanticComponentInterface<sensor_msgs::msg::NavSatFix>
```

Maybe we can use template traits to be able to handle both the cases in one single class

Reply:
I added and It worked just fine :)

Reply:
Awesome. I'm glad it worked :)